reactive/hacluster.py

```diff
@@ -0,0 +1,79 @@
+from charms.reactive import when, when_not, clear_flag, set_flag
+from charms.reactive import endpoint_from_flag
+from charmhelpers.core import hookenv
+from charms.layer.kubernetes_common import get_ingress_address
+from charmhelpers.core import unitdata
+
+db = unitdata.kv()
+
+
+@when('ha.connected', 'layer.hacluster.services_configured')
+@when_not('layer.hacluster.configured')
+def configure_hacluster():
+    """Configure HA resources in corosync"""
+    hacluster = endpoint_from_flag('ha.connected')
+    vips = hookenv.config('ha-cluster-vip').split()
+    dns_record = hookenv.config('ha-cluster-dns')
+    if vips and dns_record:
+        set_flag('layer-hacluster.dns_vip.invalid')
+        msg = "Unsupported configuration. " \
+              "ha-cluster-vip and ha-cluster-dns cannot both be set",
+        hookenv.log(msg)
+        return
+    else:
+        clear_flag('layer-hacluster.dns_vip.invalid')
+    if vips:
+        for vip in vips:
+            hacluster.add_vip(hookenv.application_name(), vip)
+    elif dns_record:
+        ip = get_ingress_address()
```

Comment:
Missing the `endpoint_name` arg to [get_ingress_address](https://github.com/juju-solutions/kubernetes/blob/87aad27c4781701a03582b9e64e1f589ad815719/cluster/juju/layers/kubernetes-common/lib/charms/layer/kubernetes_common.py#L151) here.

What is this IP used for? Would [unit_public_ip](https://github.com/juju/charm-helpers/blob/c59a6b3f14498b2ba63b6df3fe96d59cef57f391/charmhelpers/core/hookenv.py#L821) be more appropriate?

Reply:
This code was ripped from the existing usage of hacluster without a layer. I would think that a public address would be the best option here as well.

Reply:
hmm, looks like the vault charm is using a private address: https://github.com/openstack/charm-vault/blob/3b947315200f295651eed07f5ca80a65abca459e/src/reactive/vault_handlers.py#L369

Reply:
Using get_ingress_address with an endpoint defined in the layer options